components/withNavBar.tsx

```diff
@@ -6,14 +6,14 @@ import type { FC } from 'react';
 
 import NavBar from '@/components/Containers/NavBar';
 import WithBanner from '@/components/withBanner';
-import { useClientContext, useSiteNavigation } from '@/hooks';
-import { useRouter } from '@/navigation.mjs';
+import { useSiteNavigation } from '@/hooks';
+import { usePathname, useRouter } from '@/navigation.mjs';
 import { availableLocales } from '@/next.locales.mjs';
 
 const WithNavBar: FC = () => {
   const { navigationItems } = useSiteNavigation();
   const { resolvedTheme, setTheme } = useTheme();
-  const { pathname } = useClientContext();
+  const pathname = usePathname();
```

Comment:
Use the client context hook. If by any chance you're using the regular `usePathname` because of the value being different from the client context hook, then something is wrong.